src/ClassValidatorMiddleware.ts

```diff
@@ -20,24 +20,36 @@ export class ClassValidatorMiddleware<T extends object>
   /** The logger used in the module */
   private readonly logger: IDebugger;
 
-  private readonly classType: ClassType<T>;
+  private readonly bodyClassType: ClassType<T> | undefined;
+  private readonly queryClassType: ClassType<T> | undefined;
 
   /** Creates a new JSON error handler middleware */
   constructor(options: IMiddlewareOptions<T>) {
     this.logger = debugFactory("middy-middleware-class-validator");
     this.logger("Setting up ClassValidatorMiddleware");
-    this.classType = options.classType;
+    this.bodyClassType = options.classType || options.bodyClassType;
```

Comment:
What happens if both `bodyClassType` and `classType` are set? I would recommend to throw an error in case `classType` and one or both of the more specific ones are set, as that is not intended usage.

Reply:
You could adapt the `isMiddlewareOptions` from below to achieve this.